entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md

```diff
@@ -1182,6 +1182,68 @@ protected override void Up(MigrationBuilder migrationBuilder)
 }
 ```
 
+<a name="concurrent-migrations"></a>
+
+### Protection against concurrent migrations
+
+EF9 introduces a locking mechanism when executing migrations to protect against multiple migrations happening simultaneously, which could leave the database in a corrupted state. This should be a rare occurrence when migrations are deployed to the production environment using [recommended methods](/ef/core/managing-schemas/migrations/applying#sql-scripts), however it could happen more often if migrations are applied at runtime using the `IMigrator.Migrate()` method.
```

Comment:
```suggestion
EF9 introduces a locking mechanism to protect against multiple migrations executions happening simultaneously, as that could leave the database in a corrupted state. This should be a rare occurrence when migrations are deployed to the production environment using [recommended methods](/ef/core/managing-schemas/migrations/applying#sql-scripts), however it is more likely to happen if migrations are applied at runtime using the [`DbContext.Database.Migrate()`](/dotnet/api/microsoft.entityframeworkcore.relationaldatabasefacadeextensions.migrate) method.
```

Reply:
Re "This should be a rare occurrence when migrations are deployed to the production environment using recommended methods" - shouldn't it never happen at all if you're using the recommended methods? Like if you're using migration bundles or SQL scripts, you're in control and are presumably never running in them in parallel?